frontend/model/contracts/group.js

```diff
@@ -1015,10 +1015,42 @@ sbp('chelonia/defineContract', {
         mincomeAmount: x => typeof x === 'number' && x > 0,
         mincomeCurrency: x => typeof x === 'string'
       }),
-      process ({ meta, data }, { state }) {
+      process ({ contractID, meta, data }, { state, getters }) {
+        const mincomeCache = 'mincomeAmount' in data ? state.settings.mincomeAmount : null
+
         for (const key in data) {
           Vue.set(state.settings, key, data[key])
         }
+
+        if (mincomeCache !== null) {
+          // if mincome has changed and the user has set up their income details,
+          // emit 'MINCOME_CHANGED' notification, which prompts the user to choose whether they want to update income details or not.
+          const { loggedIn } = sbp('state/vuex/state')
```

Comment:
Can't access state outside of contract state inside of a `process` function. `sbp('state/vuex/state')` can only be called inside of a sideEffect.